bk-deploy-manager/deploy_manager.py

```diff
@@ -48,15 +48,17 @@ def get_candidates():
 
     for n in nodes["data"]:
         node_id = n["id"]
+        registration_event = n.get("registration_event")
+        wes_deploy_event = n.get("wes_deploy_event")
         # print("id: "+node_id)
         # print("wes_deploy_event: "+n["wes_deploy_event"])
         if n.get("beehive") in ["", None]:
             logging.info(f"node {node_id} does not belong to a beehive")
             continue
 
```

Comment:
You may want to add a check here before the deploy event check like:

```python
if registration_event in ["", None]:
    logging.info("node %s is not registered", node_id)
    continue
```

This _shouldn't_ ever happen... but it may cause the parseTime below to fail if that's None.